optimum/intel/openvino/modeling_decoder.py

```diff
@@ -335,9 +333,21 @@ class OVModelForCausalLM(OVBaseDecoderModel, GenerationMixin):
             checkpoint="gpt2",
         )
     )
+
+    def create_infer_request(self):
+        if self.compiled_model is None:
+            super().compile()        
+        return self.compiled_model.create_infer_request()
+
+    def generate(self, *args, **kwargs):
+        if kwargs.get("infer_request") is None:
+            kwargs["infer_request"] = self.create_infer_request()
+        return super().generate(*args, **kwargs)
+
     def forward(
         self,
         input_ids: torch.LongTensor,
+        infer_request: openvino.runtime.InferRequest,
```

Comment:
Is it truly backward compatible to add another positional argument to `forward` that is exposed externally and is can be potentially used in some not so trivial examples bypassing generate method? Is it safer to add it to `kwargs` instead?

Reply:
In my opinion, the safest way would be to pass it as keyword-only argument. It is more descriptive and clear approach than kwargs.